strawberry/experimental/pydantic/object_type.py

```diff
@@ -144,13 +162,16 @@ def wrap(cls):
         # Sort fields so that fields with missing defaults go first
         # because dataclasses require that fields with no defaults are defined
         # first
-        missing_default = []
-        has_default = []
-        for field in all_model_fields:
-            if field[2].default is dataclasses.MISSING:
-                missing_default.append(field)
+        missing_default: List[Tuple[str, Any, dataclasses.Field]] = []
+        has_default: List[Tuple[str, Any, dataclasses.Field]] = []
+        for model_field in all_model_fields:
+            if (
+                model_field[2].default is dataclasses.MISSING
+                and model_field[2].default_factory is dataclasses.MISSING # type: ignore
+            ):
+                missing_default.append(model_field)
             else:
-                has_default.append(field)
+                has_default.append(model_field)
```

Comment:
why do we need this change? did we miss some cases?

Reply:
yes. i had to add  and model_field[2].default_factory is dataclasses.MISSING because it would only sort out defaults set by a typical default. and it wouldn't sort those defaults added by a default_factory.

i should add a test to prevent such as regression.


also, had to rename field -> model_field because i used field in the early for loop, and mypy was erroring out

Reply:
the rename sounds good! shall we add a test for these cases too? 😊

Reply:
added!